scripts/west_commands/runners/nrfutil.py

```diff
@@ -40,27 +40,31 @@ def do_create(cls, cfg, args):
                                    recover=args.recover)
 
     def _exec(self, args):
-        try:
-            out = self.check_output(['nrfutil', '--json', 'device'] + args)
-        except subprocess.CalledProcessError as e:
-            # https://docs.python.org/3/reference/compound_stmts.html#except-clause
-            cpe = e
-            out = cpe.stdout
-        else:
-            cpe = None
-        finally:
-            # https://github.com/ndjson/ndjson-spec
-            out = [json.loads(s) for s in
-                   out.decode(sys.getdefaultencoding()).split('\n') if len(s)]
-            self.logger.debug(f'output: {out}')
-            if cpe:
-                if 'execute-batch' in args:
-                    for o in out:
-                        if o['type'] == 'batch_end' and o['data']['error']:
-                            cpe.returncode = o['data']['error']['code']
-                raise cpe
-
-        return out
+        jout_all = []
+        cmd = ['nrfutil', '--json', 'device'] + args
+        with subprocess.Popen(cmd, stdout=subprocess.PIPE) as p:
```

Comment:
Please log the command and allow for dry runs like here:
https://github.com/zephyrproject-rtos/zephyr/blob/99f62a749cbbb0da676b6320c2c9492e3914b2d1/scripts/west_commands/runners/core.py#L746-L748